rct229/rules/section5.py

```diff
@@ -97,3 +104,213 @@ def rule_check(self, context, calc_vals, data=None):
 
 
 # ------------------------
+
+
+class Section5Rule8(RuleDefinitionListIndexedBase):
+    """Rule 8 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelop)"""
+
+    def __init__(self):
+        super(Section5Rule8, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section5Rule8.BuildingRule(),
+            index_rmr="baseline",
+            id="5-8",
+            description="Baseline below-grade walls shall match the appropriate assembly maximum C-factors in Table G3.4-1 through G3.4-8.",
+        )
+
+    class BuildingRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section5Rule8.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                rmr_context="",
+            )
+
+        def get_calc_vals(self, context, data=None):
+            # Climate zone should have only one in RMR
+            climate_zone = find_exactly_one("$..climate_zone", context.baseline)
+            # QNS will there be multiple building scenario?
+            building = find_one("$..buildings[*]", context.baseline)
+            scc_dictionary_b = get_surface_conditioning_category_dict(
+                climate_zone, building
+            )
+            # QNS this code is likely to get all the surfaces under the RMR files regardless of the buildings.
+            surfaces_b = find_all("$..zones[*].surfaces[*]", building)
+            # build-in manual check flag.
+            calc_val = {}
+            failing_surface_c_factor_ids = []
+            mix_surface_c_factor_ids = []
+
+            for surface_b in surfaces_b:
+                if get_opaque_surface_type(surface_b) == "BELOW-GRADE WALL":
```

Comment:
We should export the opaque surface types from `get_opaque_surface_type.py` as constants and import them here.

Reply:
Imported